### DISCUSSION TITLE
Command mvn install failed

### DISCUSSION THREAD

**Discussion by Moosigema** [keycloakify/keycloakify#650] [Q&A]
Hi, 
When I try to build my theme, I have this error :
```
Error: Command failed: mvn install -Dmaven.repo.local="/path/to/my/project/node_modules/.cache/keycloakify/maven/keycloakify-theme/.m2"

    at ChildProcess.exithandler (node:child_process:419:12)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1091:16)
    at ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  code: 1,
  killed: false,
  signal: null,
  cmd: 'mvn install -Dmaven.repo.local="/path/to/my/project/node_modules/.cache/keycloakify/maven/keycloakify-theme/.m2"'
}
```

I tried on locally on my mac, and also on docker, but still the same error. 
For docker, I used the custom image from the doc [https://docs.keycloakify.dev/importing-your-theme-in-keycloak](https://docs.keycloakify.dev/importing-your-theme-in-keycloak)

But when I run `npx keycloakify start-keycloak` it works perferctly. 

I have no clue how to fix it.

**Comment by garronej** [comment]
Hello @Moosigema,  

What version of Keycloakify are you using?  

Can you show me your package.json?

**Reply by Moosigema** [reply]
I'm using keycloakify@10.1.0 

```
{
    "name": "keycloakify-starter",
    "version": "0.0.0",
    "description": "Starter for Keycloakify 10",
    "type": "module",
    "scripts": {
        "dev": "vite",
        "build": "tsc && vite build",
        "build-keycloak-theme": "npm run build && keycloakify build",
        "storybook": "storybook dev -p 6006",
        "format": "prettier . --write"
    },
    "license": "MIT",
    "keywords": [],
    "dependencies": {
        "keycloakify": "^10.1.0",
        "react": "^18.2.0",
        "react-dom": "^18.2.0"
    },
    "devDependencies": {
        "storybook": "^8.1.10",
        "@storybook/react": "^8.1.10",
        "@storybook/react-vite": "^8.1.10",
        "@types/react": "^18.2.43",
        "@types/react-dom": "^18.2.17",
        "@typescript-eslint/eslint-plugin": "^6.14.0",
        "@typescript-eslint/parser": "^6.14.0",
        "@vitejs/plugin-react": "^4.2.1",
        "eslint": "^8.55.0",
        "eslint-plugin-react-hooks": "^4.6.0",
        "eslint-plugin-react-refresh": "^0.4.5",
        "eslint-plugin-storybook": "^0.8.0",
        "prettier": "3.3.1",
        "typescript": "^5.2.2",
        "vite": "^5.0.8"
    },
    "engines": {
        "node": "^18.0.0 || >=20.0.0"
    }
}
```

I've checked if maven package was in node_modules, but it is not

**Reply by Moosigema** [reply]
When I run the failed command with more logs, I got this error :
```
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/path/to/my/project). Please verify you invoked Maven from the correct directory. -> [Help 1]
org.apache.maven.lifecycle.MissingProjectException: The goal you specified requires a project to execute but there is no POM in this directory (/path/to/my/project). Please verify you invoked Maven from the correct directory.
```

**Comment by garronej** [comment]
Try removing your node_modules and try again.  
Try deleting your project a do a fresh clone to be 100% sure.  

If it still dosen't work.  
What package manager are you using?  

Does this work?  
```bash
git clone https://github.com/keycloakify/keycloakify-starter
cd keycloakify-starter
yarn
yarn build-keycloak-theme
```

**Reply by Moosigema** [reply]
I removed node_modules and tried again, but didn't work. 
I have recreated a new project and it seems to work fine. 

Thank you very much @garronej for your quick response

**Comment by Moosigema** [comment]
If someone has the same issue : 
my issue came from `vite.config.ts` file, when I combine `accountThemeImplementation: "Multi-Page"` and `keycloakVersionTargets` options. I disabled version 24 and those below and it works.

**Comment by garronej** [comment]
@Moosigema,  
Ok I think I understand what was going on.  
But I don't understand why you didn't get the type error 🤔   

The Keycloak version range targets are different when you have a Mulit-Page account theme:  

<img width="648" alt="image" src="https://github.com/user-attachments/assets/9189a343-1e90-4aec-99d8-8883c8a74bb5">

Compared to when you have no account theme or a Single-Page account theme:  

<img width="981" alt="image" src="https://github.com/user-attachments/assets/29992335-577b-460c-bab3-b2a9e95d38f4">

@Moosigema What was the value of your `keycloakVersionTargets` in your vite.config.ts?

**Reply by Moosigema** [reply]
When I got this error the value was :
```
keycloakVersionTargets:{
                "21-and-below":false,
                "23":"my-theme.jar",
                "24":"my-theme.jar",
                "25-and-above": "my-theme-latest.jar",
            }
```

And now it is :
```
keycloakVersionTargets:{
                "21-and-below":false,
                "23":false,
                "24":false,
                "25-and-above": "my-theme.jar",
            }         
```

But I just realized that it was because I had the same theme name. 
I have just changed to 
```
keycloakVersionTargets:{
                "21-and-below":false,
                "23":"my-theme-23.jar",
                "24":"my-theme-24.jar",
                "25-and-above": "my-theme.jar",
            }
```
And it works perfectly
